extensions/dicom-pdf/src/pdfConfig.js

```diff
@@ -0,0 +1,23 @@
+import id, { ViewportModuleId, SOPClassHandlerId } from './id.js';
+import ConfigPoint from 'config-point';
+
+// Register the handlers with the default mode settings
+const { modeConfig } = ConfigPoint.register({
+  modeConfig: {
```

Comment:
@swederik and @sedghi This file shows how the an extension can register itself with the default mode settings.  Registering that way also registers with any mode that inherits from default.
The idea is that it defines the viewports object that gets injected into the viewports list, and then is accessible directly inside the modes instance.  This does NOT prevent the mode from customizing itself to exclude defaults, or to handle them in some other way.
This also injects the get sop class handler configuration - not the sop class handler itself, but just the configuration,
although it would be possible to inject the actual handler as well, which might be easier than loading it via a service.

Reply:
Hmm, interesting.

My wonder is what if I want to use just the sop class handler, but not the viewport at all from the extension (I want to use my own 100% custom viewport for this use case).

Currently with the way modes work I would just create a mode config that uses the SOPClassHandler and then my own viewport from a different extension.

I see how defaults could be useful, but I really don't want a user to have to define a ConfigPoint configuration if they aren't doing something complex (i.e. just to override defaults), so it would be great if we could still just define a mode as is (just as a dumb object that works via composition).

Then again, it would be really cool if you could add an extension and easily set up default behaviour...

Could you comment on how we approach these points in this model?

Reply:
Another thought is that a limitation of this approach is that mode schema is leaking into extension schema.

If mode schema changes, extensions now all have to change. Currently, extensions only have to oblige to comply to the extension contract (getPanelModule, getSopClassHandlerModule, etc), and the mode schema is decoupled so that it can change how it consumes these extensions freely without everyone's extensions changing.

This will be important in the OHIF vision of a large scale ecosystem, whereby many if not all of the best extensions are third party.